### PR TITLE
github actions build

### PR DESCRIPTION
**Pull request by andiwand**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          mkdir -p build
+          cd build
+          cmake -DCMAKE_INSTALL_PREFIX:PATH=install ..
+      - name: make
+        run: cd build && cmake --build .
+
+      - name: test
+        run: build/test/odr_test
+
+      - name: install
```
Comment (TomTasche) [inline .github/workflows/build.yml:42]: I think I don't understand what this action is meant to do. why are you building the code twice and uploading binaries afterwards?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        uses: actions/checkout@v2
+
+      - name: ubuntu install ccache
+        if: startsWith(matrix.os, 'ubuntu')
+        run: sudo apt install ccache
+      - name: macos install ccache
+        if: startsWith(matrix.os, 'macos')
+        run: brew install ccache
+
+      - name: ccache
```
[the remark this answers is not in the document]
Comment (andiwand) [reply .github/workflows/build.yml:25]: ccache speeds up the build like 5x 😱

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        uses: actions/cache@v1
+        with:
+          path: ~/.ccache
+          key: ${{ matrix.os }}-${{ matrix.arch }}-${{ matrix.compiler }}
+
+      - name: cmake
+        run: |
+          mkdir -p build
+          cd build
+          cmake -DCMAKE_INSTALL_PREFIX:PATH=install ..
```
[the remark this answers is not in the document]
Comment (andiwand) [reply .github/workflows/build.yml:35]: in cmake there is also a install phase. then you can `make install` and only place the binaries you need.

File: .github/workflows/build.yml (hunk 1 above)
Comment (andiwand) [reply .github/workflows/build.yml:42]: i build it once and install it afterwards into the `install` directory

Comment (TomTasche) [reply .github/workflows/build.yml:42]: are these binaries actively used or just for archiving purposes?

Comment (andiwand) [reply .github/workflows/build.yml:42]: we would use them for testing

**Pull request merged**